### PR TITLE
argbash: init at 2.8.1

### PR DESCRIPTION
**Pull request by rencire**
###### Motivation for this change
Add new package: [argbash](https://github.com/matejak/argbash)

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

File: pkgs/development/tools/misc/argbash/default.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{ stdenv, fetchFromGitHub, autoconf }:
+
+stdenv.mkDerivation rec {
+  name = "argbash";
```
Comment (mmahut) [inline pkgs/development/tools/misc/argbash/default.nix]: Please use ``pname``.

**Comment by mmahut** [comment]
@GrahamcOfBorg build argbash

File: pkgs/development/tools/misc/argbash/default.nix (hunk 1 above)
Comment (rencire) [reply pkgs/development/tools/misc/argbash/default.nix]: oops, thanks!

**Comment by mmahut** [comment]
@GrahamcOfBorg build argbash

File: pkgs/development/tools/misc/argbash/default.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  version = "2.8.1";
+
+  src = fetchFromGitHub {
+    owner = "matejak";
+    repo = "argbash";
+    rev = "${version}";
+    sha256 = "0jzfr2w5pza33mvw6hq6rm1vjgrfy95dp4421403jnji3ziwgf0w";
+  };
+
+  sourceRoot = "${src}/resources";
```
Comment (nlewo) [inline pkgs/development/tools/misc/argbash/default.nix]: minor: i think a relative path is sufficient

File: pkgs/development/tools/misc/argbash/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+  nativeBuildInputs = [ autoconf ];
+
+  makeFlags = [ "PREFIX=$(out)" ];
+
+  meta = with stdenv.lib; {
+    description = "Bash argument parsing code generator";
+    homepage = "https://argbash.io/";
```
Comment (nlewo) [inline pkgs/development/tools/misc/argbash/default.nix]: No quote around the url.

**Review by nlewo — CHANGES_REQUESTED** [review]
It seems you need to update the hash since it fails on ofBorg. Maybe you already have a derivation with this hash in your cache.

File: pkgs/development/tools/misc/argbash/default.nix (hunk 3 above)
Comment (mmahut) [reply pkgs/development/tools/misc/argbash/default.nix]: Please keep the quote around the url, as per https://github.com/NixOS/rfcs/pull/45.

File: pkgs/development/tools/misc/argbash/default.nix (hunk 2 above)
Comment (rencire) [reply pkgs/development/tools/misc/argbash/default.nix]: Tried `./resources` and `resources` and it didn't work.  Seems like `${src}` is required?

**Pull request merged**